patina_dxe_core/src/dispatcher.rs

```diff
@@ -41,6 +42,8 @@ use crate::{
     tpl_mutex::TplMutex,
 };
 
+static HOB_LIST: Once<&'static hob::HobList<'static>> = Once::new();
```

Comment:
General question: could you just reject any duplicate FvNameGuid without recourse to the hob list? I think that's what the EDK2 core does...